repose-aggregator/commons/utilities/src/main/scala/org/openrepose/commons/utils/servlet/http/HttpServletResponseWrapper.scala

```diff
@@ -0,0 +1,127 @@
+ * adds. In either case the headers will be written directly to the underlying response as they are added. A mode of
+ * ResponseMode.MUTABLE will enable the append, replace, and remove header methods. In mutable mode commitToResponse
+ * must be called when all changes to the response are done otherwise none of the headers will written to the underlying
+ * response.
+ *
+ * When working with the body a mode of ResponseMode.PASSTHROUGH will have all calls go directly to the provided output
+ * stream, which in the case of one of the constructors is the original response's output stream. A mode of
+ * ResponseMode.READONLY will see us wrapping the provided output stream in one that will record everything that goes
+ * through but still passing the data along to the provided stream. This can be accessed by calling
+ * getOutputStreamAsInputStream. If you desire to see the output after it went through a provided output stream you can
+ * use our wrapping stream xxxxxxx yourself by first wrapping the original response's output stream in our wrapper and
+ * then wrapping our stream with your own, and passing that in and setting the mode to ResponseMode.PASSTHROUGH.
+ * ResponseMode.MUTABLE behaves similiarly, but will not write to the underlying output stream until commitToResponse is
+ * called. Additionally, you can call setOutput with an input stream which we will wrap in an output stream which will
+ * be used to write to the underlying output stream when commitToResponse is called.
+ *
+ * @constructor the main constructor to be used for this class
+ * @param originalResponse the response to be wrapped
+ * @param headerMode the read/write-ability mode of headers
+ * @param bodyMode the read/write-ability mode of the body
+ * @param desiredOutputStream the underlying output stream to be presented callers of getOutputStream, maybe wrapped
+ *                            depending on mode selection of the body
+ */
+class HttpServletResponseWrapper(originalResponse: HttpServletResponse, headerMode: ResponseMode, bodyMode: ResponseMode, desiredOutputStream: ServletOutputStream)
+  extends javax.servlet.http.HttpServletResponseWrapper(originalResponse)
+  with HeaderInteractor {
+
+  /**
+   * This constructor chains to the main constructor using the original responses output stream  as the last arguement.
+   *
+   * @constructor the constructor to use when you don't plan on using a custom stream for processing
+   * @param originalResponse the response to be wrapped
+   * @param headerMode the mode to use for header accessiblity
+   * @param bodyMode the mode to use for body accessibility
+   */
+  def this(originalResponse: HttpServletResponse, headerMode: ResponseMode, bodyMode: ResponseMode) =
+    this(originalResponse, headerMode, bodyMode, originalResponse.getOutputStream)
+
+  override def getHeaderNamesList: util.List[String] = ???
+
+  override def addHeader(headerName: String, headerValue: String, quality: Double): Unit = ???
```

Comment:
It's part of the base class, so it didn't get stubbed out. We will do an implementation though.